lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -2228,8 +2228,55 @@ namespace Js
         }
 
         ScriptContext* scriptContext = pThis->type->GetScriptContext();
-
         ApiError error = ApiError::NoError;
+        const char16 *pThisSz = pThis->GetSz();
+        charcount_t pThisLength = pThis->GetLength();
+
+        bool isAscii = true;
```

Comment:
Looks like another case where having `JavascriptString`s know whether they contain only 7 bit ascii or not would be helpful, similar to when I was trying out utf8 strings.

Reply:
Correct. I was thinking, the parser must know or easily be able to know if literal strings are ASCII, right? If so, I feel like, since JavascriptString objects are immutable, you could easily add `JavascriptString::IsAscii()` that could either be told up front on construction or compute the result and cache it later. But yes, theres lots of places in Intl code that can benefit from ASCII too. Especially because when you know a string is ASCII-only, you dont have to worry about character indexes and lengths compared to UTF-16, since the length of an ascii string will always be the same as its' length in char16s.